prometheus/src/cluster_metrics.rs

```diff
@@ -25,18 +39,105 @@ fn get_vote_state(bank: &Bank, vote_pubkey: &Pubkey) -> Option<ValidatorVoteInfo
     let vote_state = vote_account.vote_state();
     let vote_state = vote_state.as_ref().unwrap_or(&default_vote_state);
 
+    let identity = vote_state.node_pubkey;
+
+    let validator_info: ValidatorInfo = match get_validator_info_accounts(bank) {
+        None => ValidatorInfo {
+            name: "unknown".to_string(),
+        },
+        Some(mut info_map) => match info_map.remove(&identity) {
+            None => ValidatorInfo {
+                name: "unknown".to_string(),
+            },
+            Some(info) => info,
+        },
+    };
```

Comment:
We can compose `f: T -> Option<U>` and `g: U -> Option<V>` with `f(x).and_then(g)`:
```suggestion
    let validator_info: ValidatorInfo = get_validator_info_accounts(bank)
        .and_then(|info_map| info_map.get(&identity).cloned())
        .unwrap_or(ValidatorInfo { name: "unknown".to_string() })
```

If we now want to make it an `Option<ValidatorInfo>`, then the `.unwrap_or` part simply goes away.